Objective-C/CBLReplicator.mm

```diff
@@ -699,9 +729,21 @@ - (void) resolveConflict: (CBLReplicatedDocument*)doc {
 - (void) _resolveConflict: (CBLReplicatedDocument*)doc {
     CBLLogInfo(Sync, @"%@: Resolve conflicting version of '%@'", self, doc.id);
     NSError* error = nil;
-    if (![[_config.database defaultCollectionOrThrow] resolveConflictInDocument: doc.id
-                                                           withConflictResolver: _config.conflictResolver
-                                                                          error: &error]) {
+    CBLCollection* collection = nil;
+    if (doc.collection) {
+        collection = [_config.database collectionWithName: doc.collection scope: doc.scope error: &error];
```

Comment:
The collection should be gotten from _config.collectionConfigs. I think you could set callbackContext in C4ReplicationCollection with the CBLCollection object so that you could get the CBLCollection object back from CBLReplicatedDocument.

https://github.com/couchbase/couchbase-lite-core/blob/master/C/include/c4ReplicatorTypes.h#L118